APKBUILD.5

```diff
@@ -0,0 +1,488 @@
+.Li 0 .
+
+.It Cm pkgdesc
+Specifies what the package contains.
+.Cm pkgdesc
+must be 128 characters or less, and should concisely describe what actions the
+software or items being packaged will allow the user to perform.  For example,
+.Li Dq Fully-featured word processor with spell check and many plugins
+would be a sufficient
+.Cm pkgdesc
+for AbiWord.
+
+.It Cm url
+Specifies the Web address of the package's upstream.  This allows users and
+future maintainers to find documentation, release information, and contact
+information for the package.  If no Web address is available for the package,
+you must set
+.Cm url
+to an empty string ("").
+
+.It Cm license
+Specifies the license under which the package is distributed.  The value
+provided must match a SPDX license identifier.
+
+.It Cm source
+Specifies the location of both local and remote source files used to build the
+package.  Typically, the remote source file(s) or archive(s) is specified,
+followed by any local patches, install scripts, configuration files, or other
+necessary files.
+.El
+
+
+.Ss Optional Variables
+The following variables are not required, but may be set in any
+.Nm
+file:
+
+.Bl -tag -width Ds
+.It Cm arch
+Specifies the architectures for which the package may be built.  It is highly
+recommended that you set this variable to "all" if the package is portable.
+
+You may use "noarch" if the package does not contain any architecture-specific
+binary files - that is, any files that are compiled for the target only.  Such
+packages may include pure Python packages, shell script packages, and JARs.  If
+you are not sure what this means, using "all" is safe.
+
+.It Cm depends
+Specifies the run-time dependencies of the package.  The
+.Xr abuild 1
+utility will automatically scan the resultant package for shared library (.so)
+dependencies; do not specify them here.
+
+.It Cm install
+Specifies install scripts for the package, if any.  See
+.Sx Install Scripts
+for more information about install scripts.
+
+.It Cm install_if
+Specifies a condition when
+.Xr apk 8
+should automatically install the package (or subpackage).  For instance, the
+OpenRC subpackages set
+.Cm install_if
+to
+.Li Dq $pkgname=$pkgver openrc
+which means that the OpenRC subpackage will be automatically installed if the
+origin package and OpenRC are both installed on the same computer.
+
+.It Cm makedepends
+Specifies build dependencies for the package.
+
+.It Cm pkggroups
+Specifies a space-separated list of login groups to create during build-time.
+Note that you will need to create the login groups in a pre-install install
+script as well; see
+.Sx Install Scripts
+for more information about install scripts.
+
+.It Cm pkgusers
+Specifies a space-separated list of user logins to create during build-time.
+Note that you will ned to create the user logins in a pre-install install
+script as well; see
+.Sx Install Scripts
+for more information about install scripts.
+
+.It Cm provides
```

Comment:
Some of the wiki had sentence fragments or bad grammar/typos.  (I do not have a wiki account.)

Also, the wiki does not have difference between required and optional parameters (when I first wrote an APKBUILD back in 2012, I thought they were all required).  This manual is a completely new document that I have written, based on the information I learned from the wiki, `abuild` source code, IRC/Matrix discussions, and such.  It is not just a copy of the wiki.

Reply:
I understand that it’s not just a copy of the wiki. And I really appreciate this effort to improve documentation! I’m just not happy that you (and kaniini) are not updating the current documentation at all, so creates fragmentation. I understand that MediaWiki syntax is horrible (but tbh still better than this archaic man page format…), so it may be quite annoying to work with it (at least for me it really is), but there’s still the most knowledge in one place.

Reply:
Well, the idea is actually to redirect people to rendered versions of the manpage in web documentation (the wiki could also do this).  Philosophically, we believe that Wikis are mostly used for supplemental documentation, formal documentation should exist as manpages.

Reply:
Beyond that, one of the main reasons for doing the docs cleanroom is so that we can verify that they are correct.  In other words, reconciliation of wiki docs with the cleanroom docs will ensure that all documentation is fully accurate.

Reply:
I don't mind MediaWiki syntax and I fully intend to update the wiki at some point.  It is not only differentiating formal vs supplemental documentation, but as @kaniini mentioned, trying to ensure the documentation is actually correct.  This also provides a way to read documentation while offline.

Additionally, having the documentation so close to the source will mean that if `abuild` changes in the future, there is a chance someone will update the docs when updating `abuild` itself, which means less maintenance burden.

I had an issue signing up for a wiki account some months ago (the email never came through); I can try again later.

Reply:
I think the current approach is correct.
1) write official man pages and ship those with the sources (abuild in this case)
2) autogenerate some web page based on the man page. We should have this to updated automatically (on git tag?)
3) redirect wiki to official doc. Alternatively, rewrite it and use the official man page as reference.

In any case, we should first do the official doc before fixing the wiki.